ingestion-core/src/test/resources/testdata/payload-format-duplicate-normalized-field-names-input.ndjson

```diff
@@ -0,0 +1,10 @@
+{"attributeMap":{"document_namespace":"namespace_0","document_type":"bar","document_version":"1"},"payload":{"payload":{"test_int":-2, "test.int":-2}}}
```

Comment:
Is the order in the object deterministic when read?  e.g. would `{"test_int":-2, "test.int":-3}` give consistent results?  In your comment, you said `when the "dot version" of the field precedes the "underscore version"` but I'm not understanding why the order matters

Reply:
Order is deterministic when read. Yes, `{"test_int":-2, "test.int":-3}` give consistent results.

Order matters only for map fields because if the "dot" version comes first, we will transform its value with `PubsubMessageToObjectNode.Payload#expandMapType` and insert into the normalized ("underscore") version of the field in https://github.com/mozilla/gcp-ingestion/blob/e1b318c84f189d979d75768097c51e6aaf97139a/ingestion-core/src/main/java/com/mozilla/telemetry/ingestion/core/transform/PubsubMessageToObjectNode.java#L582. Then when in the subsequent iteration we try to read this expanded value we crash.

Also compare the last line in this file with a single line in `payload-format-duplicate-normalized-map-field-input.ndjson`. These two differ only in the order of "dot" and "underscore" fields. All of the samples from this file pass through the sink, the other one throws an exception.